src/database.rs

```diff
@@ -223,9 +230,15 @@ impl EntryDiff {
                              diff.added,
                              diff.removed,
                              diff.unchanged);
+                    let mut diff_code = DiffCode::Changes;
                     for (key, entry) in entries.iter() {
-                        entry.show_diff(key, depth+1);
+                        if entry.show_diff(key, depth + 1) == DiffCode::Suspicious {
+                            diff_code = DiffCode::Suspicious;
```

Comment:
If we split this out into a separate function, I think we can make this cleaner. Maybe this is just because I'm a compiler person, but I see this as a meet operator on a semilattice. That is, `no change <meet> change = change` and `change <meet> suspicious = suspicious`, etc. Identity/bottom is no change. Then you just map `summarize` across the entries and reduce with the meet operator.